.circleci/config.yml

```diff
@@ -0,0 +1,170 @@
+# These environment variables must be set in CircleCI UI
+#
+# NPM_TOKEN - A valid NPM token for releases
+#
+# NOTE:
+# - to validate changes to this file locally using the circleci CLI tool:
+#
+#     circleci config process .circleci/config.yml
+#
+# - to try run jobs locally:
+#
+#     circleci config process .circleci/config.yml > tmp/processed.yml
+#     circleci local execute -c tmp/processed.yml --job build-nodejs-current
+#
+version: 2.1
+
+references:
+  nodejs_current: &nodejs_current "10"
+  repo_path: &repo_path ~/webextension-polyfill
+  defaults: &defaults
+    working_directory: *repo_path
+    parameters:
+      nodejs_current:
+        type: string
+        default: *nodejs_current
+
+commands:
+  run_npm_install:
+    description: install npm dependencies
+    steps:
+      - run: npm i
+
+  run_npm_build:
+    description: build project in << parameters.node_env >> mode
+    parameters:
+      node_env:
+        type: enum
+        default: production
+        enum: ["production", "test"]
+    steps:
+      - run:
+          command: npm run build --if-present
+          environment:
+            NODE_ENV: << parameters.node_env >>
+
+  run_test_minified:
+    description: rerun unit tests on minified file
+    steps:
+      - run:
+          command: npm run test-minified
+
+  run_test_bundlers:
+    description: rerun unit tests on webpack and browserify bundled files
+    steps:
+      - configure_global_npm
+      - run: npm install -g browserify webpack webpack-cli
+      - run: |
+          export PATH=$PATH:../.npm-global/bin
```

Comment:
I hope that the use of `../` won't bite us at some point (e.g. by a changed `$PWD`). If you think that this is a concern too, consider using `$HOME` or `~` or any other fixed location.

Reply:
I'm not really worried about it (hopefully these are not going to become the classic "last famous words" :-P), if the current working dir isn't the cloned repo as (configured in a reference and merged in all jobs as default base config) then most of the commands should be failing visibly.